src/test/java/uk/gov/hmcts/reform/civil/service/bundle/BundleCreationServiceTest.java

```diff
@@ -0,0 +1,162 @@
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.civil.client.BundleApiClient;
+import uk.gov.hmcts.reform.civil.config.SystemUpdateUserConfiguration;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.event.BundleCreationTriggerEvent;
+import uk.gov.hmcts.reform.civil.helpers.BundleRequestMapper;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.Party;
+import uk.gov.hmcts.reform.civil.model.ServedDocumentFiles;
+import uk.gov.hmcts.reform.civil.model.bundle.BundleCreateRequest;
+import uk.gov.hmcts.reform.civil.model.bundle.BundleCreateResponse;
+import uk.gov.hmcts.reform.civil.model.caseprogression.UploadEvidenceDocumentType;
+import uk.gov.hmcts.reform.civil.model.caseprogression.UploadEvidenceExpert;
+import uk.gov.hmcts.reform.civil.model.caseprogression.UploadEvidenceWitness;
+import uk.gov.hmcts.reform.civil.model.common.DynamicList;
+import uk.gov.hmcts.reform.civil.model.common.DynamicListElement;
+import uk.gov.hmcts.reform.civil.model.common.Element;
+import uk.gov.hmcts.reform.civil.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.civil.model.documents.Document;
+import uk.gov.hmcts.reform.civil.model.documents.DocumentType;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDetailsBuilder;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+import uk.gov.hmcts.reform.civil.service.UserService;
+import uk.gov.hmcts.reform.civil.utils.ElementUtils;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class BundleCreationServiceTest {
+
+    @InjectMocks
+    private BundleCreationService bundlingService;
+    private BundleCreationTriggerEvent event = new BundleCreationTriggerEvent(1L);
+    @Mock
+    private BundleApiClient bundleApiClient;
+    @Mock
+    private BundleCreateRequest bundleCreateRequest;
+    @Mock
+    private CaseDetailsConverter caseDetailsConverter;
+    @Mock
+    private UserService userService;
+    @Mock
+    SystemUpdateUserConfiguration userConfig;
+    @Mock
+    private CoreCaseDataService coreCaseDataService;
+
+    @Mock
+    private AuthTokenGenerator authTokenGenerator;
+    @Mock
+    private BundleRequestMapper bundleRequestMapper;
+    @Mock
+    private ObjectMapper objectMapper;
+
+    private final String testUrl = "url";
+    private final String testFileName = "testFileName.pdf";
+    CaseData caseData;
+
+    @BeforeEach
+    public void setUp() {
```

Comment:
Very long setup method. Split it in smaller blocks if possible.

Reply:
done